### PR TITLE
Update Rust Edition

### PR DESCRIPTION
**Pull request by Snowiiii**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Unreleased
+- Update from 2018 Rust Edtion to 2021
```
Comment (MarijnS95) [inline CHANGELOG.md]: Fwiw this only has meaning to code compiled within the crate, not to external crates consuming this, so there's no ~real~ reason to mention it in the changelog?

Comment (Snowiiii) [reply CHANGELOG.md]: This crate is pretty small and the changelog usally contains smaller things like Deps. updates, Many crates putting Editions changes in their changelog

Comment (msiglreith) [inline CHANGELOG.md]:
```suggestion
- Bump Rust Edition from 2018 to 2021.
```

**Review by msiglreith — COMMENTED** [review]
Thanks, a changelog entry is fine to me as the change also increases the MSRV

**Pull request merged**